core/src/main/scala/kafka/server/mirror/MirrorFetcherThread.scala

```diff
@@ -50,22 +57,176 @@ class MirrorFetcherThread(name: String,
                           quota: ReplicaQuota,
                           logPrefix: String,
                           mirrorName: String)
-  extends ReplicaFetcherThread(name, leader, brokerConfig, failedPartitions, replicaMgr, quota, logPrefix, mirrorName) {
-  /**
-   * Uses MirrorFetcherManager instead of regular ReplicaFetcherManager.
-   * This ensures proper handling of cross-cluster fetcher lifecycle when leader changes occur.
-   */
+  extends AbstractFetcherThread(name = name,
+                                clientId = name,
+                                leader = leader,
+                                failedPartitions,
+                                fetchTierStateMachine = new TierStateMachine(leader, replicaMgr, false),
+                                fetchBackOffMs = brokerConfig.replicaFetchBackoffMs,
+                                isInterruptible = false,
+                                replicaMgr.brokerTopicStats,
+                                mirrorName) {
+  this.logIdent = logPrefix
+
+  // Visible for testing
+  private[mirror] val partitionsWithNewHighWatermark = mutable.Buffer[TopicPartition]()
+
+  override def doWork(): Unit = {
+    super.doWork()
+    // Complete delayed fetch requests for consumers waiting on mirrored partitions
+    if (partitionsWithNewHighWatermark.nonEmpty) {
+      replicaMgr.completeDelayedFetchRequests(partitionsWithNewHighWatermark.toSeq)
+      partitionsWithNewHighWatermark.clear()
+    }
+  }
+
   override protected def removeFetcherForPartitions(partitions: Set[TopicPartition]): Map[TopicPartition, PartitionFetchState] = {
     replicaMgr.mirrorFetcherManager.removeFetcherForPartitions(partitions)
   }
 
-  /**
-   * Uses MirrorFetcherManager instead of regular ReplicaFetcherManager.
-   * When the remote leader changes, this method triggers the recreation logic in
-   * MirrorFetcherManager.addFetcherForPartitions, which compares the new
-   * leader's BrokerEndPoint with the existing fetcher's leader and recreates if different.
-   */
   override protected def addFetcherForPartitions(partitionAndOffsets: Map[TopicPartition, InitialFetchState]): Unit = {
     replicaMgr.mirrorFetcherManager.addFetcherForPartitions(partitionAndOffsets)
   }
+
+  // process fetched data
+  override def shouldUpdateEpochFromBatches(topicPartition: TopicPartition): Boolean = {
+    // MirrorFetcherThread always processes mirrored partitions, so always update from batches
+    true
+  }
+
+  override def processPartitionData(
```

Comment:
1. Is this override needed now? 
2. If it's not needed, maybe we can remove the changes in the `MirrorFetcherThread` and directly `extends ReplicaFetcherThread` like before?

Reply:
I think having them separated is much more clean in terms of long term maintenance at the cost of small duplication.